src/Npgsql/NpgsqlSlimDataSourceBuilder.cs

```diff
@@ -30,7 +30,7 @@ public sealed class NpgsqlSlimDataSourceBuilder : INpgsqlTypeMapper
 
     ILoggerFactory? _loggerFactory;
     bool _sensitiveDataLoggingEnabled;
-    NpgsqlTracingOptions? _tracingOptions;
+    Action<NpgsqlTracingOptionsBuilder>? _configureTracingOptionsBuilder;
```

Comment:
So the final shape of this is dependent on our conclusions, right?

Reply:
Yeah I mostly meant the multi lambda part indeed, whether that's lazy or eager.

Reply:
Right, so for now I really just followed what you had in your PR, which was a single lazily-evaluated lambda... Are we good with that for now?